sensor/tag_based_pnp_calibrator/README.md

```diff
@@ -0,0 +1,119 @@
+# tag_based_pnp_calibrator
+
+A tutorial for this calibrator can be found [here](../docs/tutorials/tag_based_pnp_calibrator.md).
+
+## Purpose
+
+The package `tag_based_pnp_calibrator` allows extrinsic calibration among camera and 3d lidar sensors used in autonomous driving and robotics.
+
+## Inner-workings / Algorithms
+
+The `tag_based_pnp_calibrator` utilizes the PnP (Perspective-n-Point) algorithm, a computer vision technique that be used to estimate the transformation between a set of correspondences of 3d and 2d points (in this case the 3d points come from the lidar and the 2d points come from the camera). For the calibration process, both `apriltag_ros` and `lidartag` packages are required, and they are executed automatically by this package's launcher files.
+
+The `apriltag_ros` package detects apriltag markers from an image and outputs the detection results. Conversely, the `lidartag` package detects lidartag markers and outputs its detection results.
+
+The `tag_based_pnp_calibrator` utilizes the detections from both `apriltag_ros` and `lidartag` packages, employing a Kalman Filter to monitor these detections. If the detections converge, the calibrator applies the SQPnP [2] algorithm provided by OpenCV to estimate the transformation between the image points from apriltag and the object points from lidartag.
+
+### Diagram
+
+Below, you can see how the algorithm is implemented in the `tag_based_pnp_calibrator` package.
+
+![segment](../docs/images/tag_based_pnp_calibrator/tag_based_pnp_calibrator.jpg)
+
+## ROS Interfaces
+
+### Input
+
+| Name                        | Type                                         | Description                                                                                  |
+| --------------------------- | -------------------------------------------- | -------------------------------------------------------------------------------------------- |
+| `{camera_info}`             | `sensor_msgs::msg::CameraInfo`               | Intrinsic parameters for the calibration cameras . `camera_info` is provided via parameters. |
```

Comment:
You copied this format from the SfM. In that case, support for several cameras was implemented. In this case, it is incorrect

Reply:
fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/dc5164fd5f105480f50f32e7ad4628871da0b194

Reply:
Parameter is a countable noun.
In addition, is is not a parameter.

Please check if the things you are writing are write before pushing. Are you using proper spell/grammar tools?

Reply:
fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/08a377f424fbd9f56a3e825592ab865fc6bbff3f

Reply:
I was using the tool before I wrote in parameter because the user can define them in the launcher.

Reply:
@vividf 
It is still wrong (in a different way this time)

Reply:
@knzo25 
| `camera_info`               | `sensor_msgs::msg::CameraInfo`               | Intrinsic parameters for the calibration camera. `camera_info` is defined in the launcher. |

Could you tell me which part is wrong? Thanks

Reply:
`Calibration camera` is a concept of the sfm where we had the external camera and the calibration cameras.
Here there is only one camera; the camera to be calibrated, so it is redundant.
Unlike the sfm, in which all the topics were passed via parameters, here the topic names are defined in the code. As it is written it is weird. If you see the launchers, there is remapping, not definition

Reply:
I fixed the first part here: https://github.com/tier4/CalibrationTools/pull/164/commits/79c9e471a8112ea5f39ea991ffdec2fc8a1ad96e

Reply:
For the description part, should I say it is remapping from the parameter "camera_info_topic"?

Reply:
When topics are created like `this->create_subscriber("topic_name",...)`, we only need to explain the contents of `topic_name`, not explain where it is defined. In particular, it is not defined in the launchers (at most, it is remaped to variable that is defined in the launcher).

The reason that it was needed in the sfm, is because the number of topics was not fixes, and as such, the list of topics was retrieved from a parameter, and thus further explanations were needed. Removing the redundant (and half incorrect part) should be applied in all the places where it is applicable.

In addition:
`Intrinsic parameters for the camera` -> `Intrinsic parameters of the camera`

Reply:
fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/f2e9bb37383fe24008b84ca421b1fed1963167d7 and https://github.com/tier4/CalibrationTools/pull/164/commits/793a581a0174ea4512d7dabed106cae1e1442271

Reply:
@vividf 
Like the Sfm, the number of calibration lidars is variable.
`calibration_pointcloud_topics` is not a topic, is a parameter

Reply:
Thanks, fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/ffcf7088589dd8ec39c75e48ad2eafcb822896ad. I changed the docker by accident so I fixed it.

Reply:
`Recommend to select the lidar that has the highest resolution.` is not grammatically correct

Reply:
fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/8d3e0512eaa6efdfedf5dc400b8f4aaf63656f89

Reply:
thanks, fixed in https://github.com/tier4/CalibrationTools/pull/164/commits/f63c2276aee893ba7b8fdaef978444505d8743cb